pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/DirectProxyHandler.java

```diff
@@ -316,6 +322,7 @@ protected void messageReceived() {
 
         @Override
         protected void handleConnected(CommandConnected connected) {
+            checkArgument(state == BackendState.Init, "Unexpected state %s. BackendState.Init was expected.", state);
```

Comment:
should we close the channel ?
other wise we will let an Unchecked Exception escape.

Reply:
a similar approach is used in the ProxyConnection class.
https://github.com/apache/pulsar/blob/c51856a699edfecc0850586befb897405a0ca9ca/pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java#L304
the `exceptionCaught` method takes care of closing the channel when any uncaught exception happens.

Reply:
thanks for checking